src/uwtools/scheduler.py

```diff
@@ -0,0 +1,117 @@
+"""
+Job Scheduling
+author: Ryan Long <ryan.long@noaa.gov>
+"""
+
+import logging
+import collections
+from typing import Dict
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s %(name)-12s %(levelname)-8s %(message)s",
+    handlers=[logging.StreamHandler()],
+)
+
+
+class JobCard(collections.UserList):
+    """represents a job card to submit to a scheduler"""
+
+    def content(self, line_separator: str = "\n") -> str:
+        """returns the formatted content of the job card"""
+        return line_separator.join(self)
+
+
+class JobScheduler(collections.UserDict):
+
+    _map = {}
+    prefix = ""
+
+    def __getattr__(self, name):
+        if name in self:
+            return self[name]
+
+    @property
+    def job_card(self) -> JobCard:
+        """returns a job card representation"""
+        return JobCard(
+            [
+                f"{self.prefix} {self._map[key] if key in self._map else key + '='}{value}"
```

Comment:
The `else` branch here prevents the use of flags that don't have values. We'd discussed `--exclusive` as an example this morning.  Would just having `else key` work here? The user would be required to provide the `=` if it is needed (precedence in the implementation of required flags)?

Reply:
It doesn't prevent values, but allows keys that are not listed in the map to not be translated (hence the adding of the '=' instead of appending to its value in the _map table.

It's not the final solution as I think we're going to have to build some intelligence into how the parsing takes the raw data -> JobCard, but it will allow tests to pass in the interim.

The tests illustrate what I mean.

Reply:
I suppose I was suggesting that instead of `else key + '='` just having `else key`. That would allow a test like this to work:

```
expected = """#SBATCH scheduler=slurm
#SBATCH --job-name=abcd
#SBATCH --exclusive"""
props = {"scheduler": "slurm", "job_name": "abcd", "--exclusive": ""}
```

When user wanted to provide a key that requires an `=` then they'd need to put that as part of the key in props. The existing test would then turn out like this:

```
expected = """#SBATCH scheduler=slurm
#SBATCH --job-name=abcd
#SBATCH extra_stuff=12345"""

    props = {"scheduler": "slurm", "job_name": "abcd", "extra_stuff=": "12345"}
```

Reply:
I understand now, thanks.  Updating.

Reply:
doesn't that mean the user needs to know what needs an `=` and what doesn't?

Reply:
Yes, the way it is now, yes.  We have to make smarter.